internal/controller/snapshot/snapshot_adapter.go

```diff
@@ -289,39 +288,53 @@ func (a *Adapter) EnsureGlobalCandidateImageUpdated() (controller.OperationResul
 		return controller.ContinueProcessing()
 	}
 
-	for _, component := range a.snapshot.Spec.Components {
-		if component.Name == a.component.Name {
-			patch := client.MergeFrom(a.component.DeepCopy())
-			a.component.Spec.ContainerImage = component.ContainerImage
-			err := a.client.Patch(a.context, a.component, patch)
+	var componentToUpdate *applicationapiv1alpha1.Component
+	var err error
+	err = retry.OnError(retry.DefaultRetry, func(_ error) bool { return true }, func() error {
```

Comment:
Basically I am fine with the code, just one question: I update the code to get component like [this](https://github.com/konflux-ci/integration-service/pull/760/files#diff-cbf7e6d58cc12a67b910b5b9108ba68d206e75cc8560ca0ab2b19e2d34415702R301) in my PR, I feel they can be reviewed together then save reviewer's time to review code.

Reply:
Since #760 will be building upon the changes in this PR, are you ok if we first merge this PR and then rebase your changes on it?

Reply:
I agree, thanks!